Code/Framework/AzCore/AzCore/Math/IntersectPoint.h

```diff
@@ -23,197 +21,48 @@ namespace AZ
     */
     namespace Intersect
     {
-        /**
-         * Compute barycentric coordinates (u,v,w) for a point P with respect
-         * to triangle (a,b,c)
-         */
-        AZ_MATH_INLINE Vector3
-        Barycentric(const Vector3& a, const Vector3& b, const Vector3& c, const Vector3& p)
-        {
-            Vector3 v0 = b - a;
-            Vector3 v1 = c - a;
-            Vector3 v2 = p - a;
-
-            float d00 = v0.Dot(v0);
-            float d01 = v0.Dot(v1);
-            float d11 = v1.Dot(v1);
-            float d20 = v2.Dot(v0);
-            float d21 = v2.Dot(v1);
-
-            float denom = d00 * d11 - d01 * d01;
-            float denomRCP = 1.0f / denom;
-            float v = (d11 * d20 - d01 * d21) * denomRCP;
-            float w = (d00 * d21 - d01 * d20) * denomRCP;
-            float u = 1.0f - v - w;
-            return Vector3(u, v, w);
-        }
 
-        /// Test if point p lies inside the triangle (a,b,c).
-        AZ_MATH_INLINE int
-        TestPointTriangle(const Vector3& p, const Vector3& a, const Vector3& b, const Vector3& c)
-        {
-            Vector3 uvw = Barycentric(a, b, c, p);
-            return uvw.IsGreaterEqualThan(Vector3::CreateZero());
-        }
-
-        /// Test if point p lies inside the counter clock wise triangle (a,b,c).
-        AZ_MATH_INLINE int
-        TestPointTriangleCCW(const Vector3& p, const Vector3& a, const Vector3& b, const Vector3& c)
-        {
-            // Translate the triangle so it lies in the origin p.
-            Vector3 pA = a - p;
-            Vector3 pB = b - p;
-            Vector3 pC = c - p;
-            float ab = pA.Dot(pB);
-            float ac = pA.Dot(pC);
-            float bc = pB.Dot(pC);
-            float cc = pC.Dot(pC);
-
-            // Make sure plane normals for pab and pac point in the same direction.
-            if (bc * ac - cc * ab < 0.0f)
-            {
-                return 0;
-            }
-
-            float bb = pB.Dot(pB);
+        //! Compute barycentric coordinates (u,v,w) for a point P with respect
+        //! to triangle (a,b,c)
+        AZ_MATH_INLINE Vector3
+        Barycentric(const Vector3& a, const Vector3& b, const Vector3& c, const Vector3& p);
 
-            // Make sure plane normals for pab and pca point in the same direction.
-            if (ab * bc - ac * bb < 0.0f)
-            {
-                return 0;
-            }
+        //! Test if point p lies inside the triangle (a,b,c).
+        AZ_MATH_INLINE bool
+        TestPointTriangle(const Vector3& p, const Vector3& a, const Vector3& b, const Vector3& c);
```

Comment:
Not really possible to fix now, but I just noticed the horrible historical inconsistency in the signatures of these functions - some are a, b, c, p and others are p, a, b, c :(